source/incqueryserver-jupyter/iqs_jupyter/core_extensions.py

```diff
@@ -386,8 +390,29 @@ def __init__(
         else:
             fields = [self.address_field]
         
-        self.box=widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
-        if auto_display: self.display()
+        self.box = widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
+        if auto_display:
+            self.display()
+
+        if login_button:
+            btn_login = interact.options(manual=True, manual_name="Login")
+
+            @btn_login
+            def login_to_iqs():
+                try:
+                    self.iqs = self.connect()
+                    server_status = requests.get(
```

Comment:
Why do we use the raw `requests` library here, instead of the `iqs` object with its generated API methods? The point of that object is to allow us to avoid having to deal with low-level http stuff.

Did you have any problems with `iqs.server_management.get_server_status()` (or if you need the error code, `iqs.server_management.get_server_status_with_http_info()`)?

Reply:
Good point! I think I did not find the correct function for that.